lib/spack/spack/spec.py

```diff
@@ -4418,7 +4420,10 @@ def old_format(self, format_string="$_$@$%@+$+$=", **kwargs):
         TODO: allow, e.g., ``$6#`` to customize short hash length
         TODO: allow, e.g., ``$//`` for full hash.
         """
-
+        tty.warn(
```

Comment:
This:
```suggestion
        warnings.warn(
```
should help a little with reducing the clutter if the user has a deprecated format (I think it emits a warning for each caller to this function, but need to double check). The net result is this:
![Screenshot from 2022-11-24 17-41-47](https://user-images.githubusercontent.com/4199709/203833404-ace9789a-6f39-4405-9d53-da1a3842d98c.png)
instead of this:
![Screenshot from 2022-11-24 17-42-41](https://user-images.githubusercontent.com/4199709/203833526-3a246cf3-3732-40a8-9eff-27486bcaf893.png)

if you temporarily revert the changes to the default `config.yaml` to check the warnings.